pgmpy/tests/test_models/test_FactorGraph.py

```diff
@@ -280,5 +280,25 @@ def test_check_model4(self):
         self.graph.add_edges_from([('a', phi3), ('c', phi3)])
         self.assertRaises(ValueError, self.graph.check_model)
 
+    def test_copy(self):
+        self.graph.add_nodes_from(['a', 'b', 'c'])
+        phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        phi2 = DiscreteFactor(['b', 'c'], [2, 2], np.random.rand(4))
```

Comment:
@lohani2280 For example we have the same statements in line 255 and 256, 237 and 238, 224 and 225.